ament_mypy/ament_mypy/main.py

```diff
@@ -0,0 +1,246 @@
+    mypy_argv.append('--show-traceback')
+    mypy_argv += paths
+    res = mypy.api.run(mypy_argv)  # type: Tuple[str, str, int]
+    return res
+
+
+def _get_xunit_content(errors: List[Match],
+                       testname: str,
+                       filenames: List[str],
+                       elapsed: float) -> str:
+    xml = textwrap.dedent("""\
+        <?xml version="1.0" encoding="UTF-8"?>
+        <testsuite
+        name="{test_name:s}"
+        tests="{test_count:d}"
+        failures="{error_count:d}"
+        time="{time:s}"
+        >
+    """).format(
+                test_name=testname,
+                test_count=max(len(errors), 1),
+                error_count=len(errors),
+                time='{:.3f}'.format(round(elapsed, 3))
+    )
+
+    if errors:
+        # report each mypy error/warning as a failing testcase
+        for error in errors:
+            pos = ''
+            if error.group('lineno'):
+                pos += ':' + str(error.group('lineno'))
+                if error.group('colno'):
+                    pos += ':' + str(error.group('colno'))
+            xml += _dedent_to("""\
+                <testcase
+                    name={quoted_name}
+                    classname="{test_name}"
+                >
+                    <failure message={quoted_message}/>
+                </testcase>
+                """, '  ').format(
+                    quoted_name=quoteattr(
+                        '{0[type]} ({0[filename]}'.format(error) + pos + ')'),
+                    test_name=testname,
+                    quoted_message=quoteattr('{0[msg]}'.format(error) + pos)
+                )
+    else:
+        # if there are no mypy problems report a single successful test
+        xml += _dedent_to("""\
+            <testcase
+              name="mypy"
+              classname="{}"
+              status="No problems found"/>
+            """, '  ').format(testname)
+
+    # output list of checked files
+    xml += "  <system-out>Checked files:{escaped_files}\n  </system-out>\n".format(
+        escaped_files=escape(''.join(['\n* %s' % f for f in filenames]))
+    )
+
+    xml += '</testsuite>\n'
+    return xml
+
+
+def _get_files(paths: List[str]) -> List[str]:
+    files = []
+    for path in paths:
+        if os.path.isdir(path):
+            for dirpath, dirnames, filenames in os.walk(path):
+                if 'AMENT_IGNORE' in filenames:
+                    dirnames[:] = []
+                    continue
+                # ignore folder starting with . or _
+                dirnames[:] = [d for d in dirnames if d[0] not in ['.', '_']]
+                dirnames.sort()
+
+                # select files by extension
+                for filename in sorted(filenames):
+                    if filename.endswith('.py'):
+                        files.append(os.path.join(dirpath, filename))
+        elif os.path.isfile(path):
+            files.append(path)
+    return [os.path.normpath(f) for f in files]
```

Comment:
Why use `normpath` here?

As the [docs](https://docs.python.org/3/library/os.path.html) say:

> This string manipulation may change the meaning of a path that contains symbolic links.

Reply:
This was copied from ament-pyflakes, which it looks like you added to avoid an issue with './' in paths https://github.com/ament/ament_lint/commit/58515dc94aa7cc0d01866d8ffc5528dd862a06e4